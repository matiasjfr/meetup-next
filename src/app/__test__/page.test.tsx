import { render, RenderOptions } from '@testing-library/react';

import RootLayout from '../layout';
import Home from '../page';

const renderWithRootLayout = (component: any, options?: RenderOptions) => {
  const tree = render(component, {
    wrapper: ({ children }: { children: React.ReactNode }) => <RootLayout>{children}</RootLayout>,
    ...options,
  });

  return { ...tree };
};

describe('Home Layout', () => {
  it('Render Home correctly', () => {
    const { container } = renderWithRootLayout(<Home />);

    expect(container).toMatchSnapshot();
  });
});
