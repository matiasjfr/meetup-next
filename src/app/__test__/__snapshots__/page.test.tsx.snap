// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Layout Render Home correctly 1`] = `
<div>
  <html
    lang="en"
  >
    <body
      class="className"
    >
      <main
        class="flex min-h-screen flex-col items-center justify-between p-24"
      >
        <div
          class="z-10 w-full max-w-5xl items-center justify-between font-mono text-sm lg:flex"
        >
          <p
            class="fixed left-0 top-0 flex w-full justify-center border-b border-gray-300 bg-gradient-to-b from-zinc-200 pb-6 pt-8 backdrop-blur-2xl dark:border-neutral-800 dark:bg-zinc-800/30 dark:from-inherit lg:static lg:w-auto  lg:rounded-xl lg:border lg:bg-gray-200 lg:p-4 lg:dark:bg-zinc-800/30"
          >
            Get started by editing 
            <code
              class="font-mono font-bold"
            >
              src/app/page.tsx
            </code>
          </p>
          <div
            class="fixed bottom-0 left-0 flex h-48 w-full items-end justify-center bg-gradient-to-t from-white via-white dark:from-black dark:via-black lg:static lg:h-auto lg:w-auto lg:bg-none"
          >
            <a
              class="pointer-events-none flex place-items-center gap-2 p-8 lg:pointer-events-auto lg:p-0"
              href="https://vercel.com?utm_source=create-next-app&utm_medium=appdir-template&utm_campaign=create-next-app"
              rel="noopener noreferrer"
              target="_blank"
            >
              By 
              <img
                alt="Vercel Logo"
                class="dark:invert"
                height="24"
                src="/vercel.svg"
                width="100"
              />
            </a>
          </div>
        </div>
        <div
          class="rounded-lg border bg-card text-card-foreground shadow-sm w-[350px]"
        >
          <div
            class="flex flex-col space-y-1.5 p-6"
          >
            <h3
              class="text-2xl font-semibold leading-none tracking-tight"
            >
              Create project
            </h3>
            <p
              class="text-sm text-muted-foreground"
            >
              Deploy your new project in one-click.
            </p>
          </div>
          <div
            class="p-6 pt-0"
          >
            <form>
              <div
                class="grid w-full items-center gap-4"
              >
                <div
                  class="flex flex-col space-y-1.5"
                >
                  <label
                    class="text-sm font-medium leading-none peer-disabled:cursor-not-allowed peer-disabled:opacity-70"
                    for="name"
                  >
                    Name
                  </label>
                  <input
                    class="flex h-10 w-full rounded-md border border-input bg-background px-3 py-2 text-sm ring-offset-background file:border-0 file:bg-transparent file:text-sm file:font-medium placeholder:text-muted-foreground focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50"
                    id="name"
                    placeholder="Name of your project"
                  />
                </div>
                <div
                  class="flex flex-col space-y-1.5"
                >
                  <label
                    class="text-sm font-medium leading-none peer-disabled:cursor-not-allowed peer-disabled:opacity-70"
                    for="framework"
                  >
                    Framework
                  </label>
                  <button
                    aria-autocomplete="none"
                    aria-controls="radix-:r0:"
                    aria-expanded="false"
                    class="flex h-10 w-full items-center justify-between rounded-md border border-input bg-transparent px-3 py-2 text-sm ring-offset-background placeholder:text-muted-foreground focus:outline-none focus:ring-2 focus:ring-ring focus:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50"
                    data-placeholder=""
                    data-state="closed"
                    dir="ltr"
                    id="framework"
                    role="combobox"
                    type="button"
                  >
                    <span
                      style="pointer-events: none;"
                    >
                      Select
                    </span>
                    <svg
                      aria-hidden="true"
                      class="h-4 w-4 opacity-50"
                      fill="none"
                      height="24"
                      stroke="currentColor"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                      stroke-width="2"
                      viewBox="0 0 24 24"
                      width="24"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="m6 9 6 6 6-6"
                      />
                    </svg>
                  </button>
                  <select
                    aria-hidden="true"
                    style="position: absolute; border: 0px; width: 1px; height: 1px; padding: 0px; margin: -1px; overflow: hidden; clip: rect(0px, 0px, 0px, 0px); white-space: nowrap; word-wrap: normal;"
                    tabindex="-1"
                  >
                    <option
                      value=""
                    />
                    <option
                      value="next"
                    >
                      Next.js
                    </option>
                    <option
                      value="sveltekit"
                    >
                      SvelteKit
                    </option>
                    <option
                      value="astro"
                    >
                      Astro
                    </option>
                    <option
                      value="nuxt"
                    >
                      Nuxt.js
                    </option>
                  </select>
                </div>
              </div>
            </form>
          </div>
          <div
            class="items-center p-6 pt-0 flex justify-between"
          >
            <button
              class="inline-flex items-center justify-center rounded-md text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 border border-input bg-background hover:bg-accent hover:text-accent-foreground h-10 px-4 py-2"
            >
              Cancel
            </button>
            <button
              class="inline-flex items-center justify-center rounded-md text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 bg-primary text-primary-foreground hover:bg-primary/90 h-10 px-4 py-2"
            >
              Deploy
            </button>
          </div>
        </div>
        <div
          class="mb-32 grid text-center lg:mb-0 lg:grid-cols-4 lg:text-left"
        >
          <a
            class="group rounded-lg border border-transparent px-5 py-4 transition-colors hover:border-gray-300 hover:bg-gray-100 hover:dark:border-neutral-700 hover:dark:bg-neutral-800/30"
            href="https://nextjs.org/docs?utm_source=create-next-app&utm_medium=appdir-template&utm_campaign=create-next-app"
            rel="noopener noreferrer"
            target="_blank"
          >
            <h2
              class="mb-3 text-2xl font-semibold"
            >
              Docs
               
              <span
                class="inline-block transition-transform group-hover:translate-x-1 motion-reduce:transform-none"
              >
                -&gt;
              </span>
            </h2>
            <p
              class="m-0 max-w-[30ch] text-sm opacity-50"
            >
              Find in-depth information about Next.js features and API.
            </p>
          </a>
          <a
            class="group rounded-lg border border-transparent px-5 py-4 transition-colors hover:border-gray-300 hover:bg-gray-100 hover:dark:border-neutral-700 hover:dark:bg-neutral-800/30"
            href="https://nextjs.org/learn?utm_source=create-next-app&utm_medium=appdir-template-tw&utm_campaign=create-next-app"
            rel="noopener noreferrer"
            target="_blank"
          >
            <h2
              class="mb-3 text-2xl font-semibold"
            >
              Learn
               
              <span
                class="inline-block transition-transform group-hover:translate-x-1 motion-reduce:transform-none"
              >
                -&gt;
              </span>
            </h2>
            <p
              class="m-0 max-w-[30ch] text-sm opacity-50"
            >
              Learn about Next.js in an interactive course with quizzes!
            </p>
          </a>
          <a
            class="group rounded-lg border border-transparent px-5 py-4 transition-colors hover:border-gray-300 hover:bg-gray-100 hover:dark:border-neutral-700 hover:dark:bg-neutral-800/30"
            href="https://vercel.com/templates?framework=next.js&utm_source=create-next-app&utm_medium=appdir-template&utm_campaign=create-next-app"
            rel="noopener noreferrer"
            target="_blank"
          >
            <h2
              class="mb-3 text-2xl font-semibold"
            >
              Templates
               
              <span
                class="inline-block transition-transform group-hover:translate-x-1 motion-reduce:transform-none"
              >
                -&gt;
              </span>
            </h2>
            <p
              class="m-0 max-w-[30ch] text-sm opacity-50"
            >
              Explore the Next.js 13 playground.
            </p>
          </a>
          <a
            class="group rounded-lg border border-transparent px-5 py-4 transition-colors hover:border-gray-300 hover:bg-gray-100 hover:dark:border-neutral-700 hover:dark:bg-neutral-800/30"
            href="https://vercel.com/new?utm_source=create-next-app&utm_medium=appdir-template&utm_campaign=create-next-app"
            rel="noopener noreferrer"
            target="_blank"
          >
            <h2
              class="mb-3 text-2xl font-semibold"
            >
              Deploy
               
              <span
                class="inline-block transition-transform group-hover:translate-x-1 motion-reduce:transform-none"
              >
                -&gt;
              </span>
            </h2>
            <p
              class="m-0 max-w-[30ch] text-sm opacity-50"
            >
              Instantly deploy your Next.js site to a shareable URL with Vercel.
            </p>
          </a>
        </div>
      </main>
    </body>
  </html>
</div>
`;
